tests/adapters/test_protocol.py

```diff
@@ -22,10 +22,177 @@
 # THE SOFTWARE.
 #
 
-from pymeasure.adapters.protocol import ProtocolAdapter
+from pymeasure.adapters.protocol import to_bytes, ProtocolAdapter
+
+from pytest import mark, raises, fixture
+
+
+@mark.parametrize("input, output", (("superXY", b"superXY"),
+                                    ([1, 2, 3, 4], b"\x01\x02\x03\x04"),
+                                    (5, b"5"),
+                                    (4.6, b"4.6")
+                                    ))
+def test_to_bytes(input, output):
+    assert to_bytes(input) == output
+
+
+def test_to_bytes_invalid():
+    with raises(TypeError):
+        to_bytes(5.5j)
 
 
 def test_protocol_instantiation():
-    a = ProtocolAdapter()
-    assert not a
+    a = ProtocolAdapter([("write", "read"), ("write_only",)])
+    assert a.comm_pairs == [("write", "read"), ("write_only",)]
     # TODO: Make this first test assert something useful
+
+
+class Test_write:
+    @mark.parametrize("command, written", (("hey Ho", b"hey Ho"),
+                                           (b"superX", "superX")
+                                           ))
+    def test_write(self, command, written):
```

Comment:
`test_write_does_not_distinguish_bytestring_from_string` is maybe a better name?
Maybe a discussion point for later if we should actually allow that...

Reply:
I just wanted to have some different inputs. In the end, to_bytes is already tested, so we do not need to test different things here.

Basically, this discussion is about whether to_bytes is applied to the comm_pairs or not. In the latter case, the protocol has to be specified in bytes.

Reply:
I think it's good to have the freedom to use either. I have not yet come up with a case where that's a bad idea.